go/arrow/csv/reader.go

```diff
@@ -76,6 +80,28 @@ func NewReader(r io.Reader, schema *arrow.Schema, opts ...Option) *Reader {
 	return rr
 }
 
+func (r *Reader) readHeader() error {
+	records, err := r.r.Read()
+	if err != nil {
+		return ErrFailedFileRead
```

Comment:
I am not sure I am completely happy with this.
on the one hand, I'd like to clearly convey at which stage (header reading) the Reader failed.
but on the other hand, I'd also like to know why it failed.

perhaps we should just use `github.com/pkg/errors.Wrapf` ?
WDYT?

Reply:
Agree, I added dependency on `github.com/pkg/errors` and used Wrapf to include the low level error, I also updated the message to make failure stage clear.